.github/workflows/main.yml

```diff
@@ -0,0 +1,30 @@
+# .github/workflows/main.yml
+
+name: CI
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+    branches:
+    - main
+
+jobs:
+  test-ruby:
+    runs-on: ${{ matrix.os }}-latest
+    strategy:
+      matrix:
+        os: [ubuntu]
+        ruby-version: ["2.6", "2.7", "3.0", "3.1", "3.2"]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Ruby ${{ matrix.ruby-version }}
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby-version }}
+          bundler-cache: true
+      - name: Build and run tests
+        run: |
+          gem install bundler
```

Comment:
I'm 99% sure that just having the above `setup-ruby` action in place with `bundler-cache: true` is going to have already run `bundle install` for you ... this whole step run section could be just the rspec run.